### PR TITLE
storyblok/react nextjs integration with visual editor

### PR DESCRIPTION
**Pull request by zelmandesign**
Storyblok with Nextjs integration. visual editor and basic components.

### CODE REVIEW

File: pages/[...slug].tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import Head from 'next/head'
+import 'bootstrap/dist/css/bootstrap.min.css'
+import styles from '../styles/Home.module.css'
+import { useStoryblokState, getStoryblokApi, StoryblokComponent } from '@storyblok/react'
```
Comment (pdobrowolskivalue) [inline pages/[...slug].tsx:4]: It is good to have correct import order: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md

File: components/Feature.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { storyblokEditable } from '@storyblok/react'
+
+const Feature = ({ blok }: any) => {
```
Comment (pdobrowolskivalue) [inline components/Feature.tsx:3]: Missing type, tricky typing - we can find end extend storyblok interface or create your own ;)
To protect data

File: pages/[...slug].tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  const storyblokApi = getStoryblokApi()
+  const { data } = await storyblokApi.get(`cdn/stories/${slug}`, sbParams)
+
+  return {
+    props: {
+      story: data ? data.story : false,
+      key: data ? data.story.id : false,
+    },
+    revalidate: 3600,
```
Comment (pdobrowolskivalue) [inline pages/[...slug].tsx:42]: Do you understand approach here?
https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration

File: pages/[...slug].tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    paths.push({ params: { slug: splittedSlug } })
+  })
+
+  return {
+    paths: paths,
+    fallback: false,
+  }
+}
+
+// export default function Home({ story: initialStory }: any) {
```
Comment (pdobrowolskivalue) [inline pages/[...slug].tsx:67]: Please move to another component or delete

File: pages/_app.tsx [hunk 5]
```diff
@@ -1,7 +1,24 @@
 import '../styles/globals.css'
-import type { AppProps } from 'next/app'
+import { storyblokInit, apiPlugin } from '@storyblok/react'
+import Page from '../components/Page'
+import Feature from '../components/Feature'
+import Grid from '../components/Grid'
+import Teaser from '../components/Teaser'
```
Comment (pdobrowolskivalue) [inline pages/_app.tsx:6]: We can import all from components

Or let's do the experiment - baseUrl + paths - you can do the research
```
{
  "compilerOptions": {
    "baseUrl": ".",
    "module": "commonjs",
    "paths": {
      "@components": "./components",
      "@utils": "./utils"
    },
    ...
  }
}
```

File: components/Feature.tsx (hunk 2 above)
Comment (zelmandesign) [reply components/Feature.tsx:3]: I've found a solution for tricky typing. 
Not sure if this is the correct approach though.
https://github.com/storyblok/storyblok-react/blob/main/playground-next/components/feature.tsx

File: pages/[...slug].tsx (hunk 3 above)
Comment (zelmandesign) [reply pages/[...slug].tsx:42]: After reading this article I do. The page is regenerated in the background when a new request comes through. The cached page is being until new page is generated.

File: components/Feature.tsx (hunk 2 above)
Comment (pdobrowolskivalue) [reply components/Feature.tsx:3]: Good. Let's try:
```
interface FeatureProps {
  blok: SbBlokData;
}
```

File: pages/_app.tsx (hunk 5 above)
Comment (zelmandesign) [reply pages/_app.tsx]: it's done

**Pull request merged**